spec/classes/redis_exporter_spec.rb

```diff
@@ -18,9 +18,18 @@
           }
         end
 
+        it { is_expected.to compile.with_all_deps }
         describe 'install correct binary' do
           it { is_expected.to contain_file('/usr/local/bin/redis_exporter').with('target' => '/opt/redis_exporter-0.11.2.linux-amd64/redis_exporter') }
         end
+        describe 'required resources' do
+          it { is_expected.to contain_group('redis-exporter') }
+          it { is_expected.to contain_prometheus__daemon('prometheus_redis_exporter') }
+          it { is_expected.to contain_service('prometheus_redis_exporter') }
+          it { is_expected.to contain_user('redis-exporter') }
+          it { is_expected.to contain_file('/opt/redis_exporter-0.11.2.linux-amd64') }
```

Comment:
This seems like it's covered by the spec on line 23 (the spec that tests that it installs the correct binary), and is not needed. The spec on line 23 is also more explicit, since it verifies that `/opt/redis_exporter....` is a symlink to `/usr/local/bin/redis_exporter` binary install location.

You can remove this condition.

Reply:
agreed, i have moved them up to 'install correct binary'